### PR TITLE
Training operator  CICD improvements

### PR DESCRIPTION
**Pull request by juliusvonkohout**
# Pull Request Template for Kubeflow manifests Issues

- Please include a summary of changes and the related issue. 
- List any dependencies that are required for this change. 
- Please delete the options that are not relevant.
- The following checklist will help you to satisfy the requirements.

## ✏️ A brief description of the changes
Training operator networkpolicies and CICD improvements

  
## ✅ Unit Test Checklist
  
  - [x] 🛠️ Make sure you have installed kustomize == 5.2.1+     
  - [x] ✍️ Have you written new tests for your core changes, as applicable?      
  - [x] 🔄 Have you successfully run existing tests with your changes ?    
  - [x] 🚀 Have you successfully run existing and new tests with your changes ?

## ✅ Contributor checklist
  - [x] All the commits have been _signed-off_  (To pass the `DCO` check)
  - [x] Submit the [Contributor License Agreements](https://cla.developers.google.com/clas) (To pass the `cla/google` check)

---     
 
> You can join the CNCF Slack and access our meetings at the [Kubeflow Community](https://www.kubeflow.org/docs/about/community/) website. Our channel on the CNCF Slack is here [**#kubeflow-platform**](https://app.slack.com/client/T08PSQ7BQ/C073W572LA2).

### CODE REVIEW

**Comment by juliusvonkohout** [comment]
@lampajr @nagar-ajay can you test this locally?

**Comment by rimolive** [comment]
Test `Build & Apply Training Operator manifests in KinD` is failling. I expect this PR will fix that test?

**Review by lampajr — COMMENTED** [review]
> Test Build & Apply Training Operator manifests in KinD is failling. I expect this PR will fix that test?

As Ricardo already pointed out there is one failing test, I tried to debug it a little bit and it looks like the condition 
```
kubectl wait --for=condition=Succeeded tfjobs tfjob-simple -n kubeflow-user-example-com --timeout 600s
```
is never met.

I tried to test it locally and it seems I am experiencing the same issue, the `tfjob/tfjob-simple` keeps remaining in `Running` state even if both pods exited with code 0:
```sh
Status:
  Conditions:
    Last Transition Time:  2024-07-01T13:14:54Z
    Last Update Time:      2024-07-01T13:14:54Z
    Message:               TFJob tfjob-simple is created.
    Reason:                TFJobCreated
    Status:                True
    Type:                  Created
    Last Transition Time:  2024-07-01T13:14:56Z
    Last Update Time:      2024-07-01T13:14:56Z
    Message:               TFJob kubeflow-user-example-com/tfjob-simple is running.
    Reason:                TFJobRunning
    Status:                True
    Type:                  Running
  Replica Statuses:
    Worker:
      Active:  2
  Start Time:  2024-07-01T13:14:54Z
Events:
  Type    Reason                   Age                    From              Message
  ----    ------                   ----                   ----              -------
  Normal  SuccessfulCreatePod      5m8s                   tfjob-controller  Created pod: tfjob-simple-worker-0
  Normal  SuccessfulCreatePod      5m8s                   tfjob-controller  Created pod: tfjob-simple-worker-1
  Normal  SuccessfulCreateService  5m8s                   tfjob-controller  Created service: tfjob-simple-worker-0
  Normal  SuccessfulCreateService  5m8s                   tfjob-controller  Created service: tfjob-simple-worker-1
  Normal  ExitedWithCode           4m11s (x2 over 4m11s)  tfjob-controller  Pod: kubeflow-user-example-com.tfjob-simple-worker-0 exited with code 0
  Normal  ExitedWithCode           4m11s                  tfjob-controller  Pod: kubeflow-user-example-com.tfjob-simple-worker-1 exited with code 0
```

I don't know exactly what could be the issue here, but for sure the came condition was met before https://github.com/kubeflow/manifests/actions/runs/9675816305

**Comment by juliusvonkohout** [comment]
We have to rebase after https://github.com/kubeflow/manifests/pull/2779 is merged.

**Comment by juliusvonkohout** [comment]
@kubeflow/wg-training-leads can you help? I can probably only work on this after the 1.9 release.

File: tests/gh-actions/install_multi_tenancy.sh [hunk 1]
```diff
@@ -7,3 +7,6 @@ kubectl -n kubeflow wait --for=condition=Ready pods -l kustomize.component=profi
 
 echo "Installing Multitenancy Kubeflow Roles"
 kustomize build common/kubeflow-roles/base | kubectl apply -f -
+
+echo "Installing Multitenancy Network policies"
+kustomize build common/networkpolicies/base | kubectl apply -f -
```
Comment (juliusvonkohout) [inline tests/gh-actions/install_multi_tenancy.sh:12]: @biswajit-9776 Lets salvage what  we have in this PR except for the training-operator changes and add below the networkpolicies the static and dynamic PSS stuff in your own PR.

I will probably close this PR here afterwards and reopen another one for the training operator test restructuring

**Comment by hansinikarunarathne** [comment]
I did the requested changes in the PR [https://github.com/kubeflow/manifests/pull/2818](2818) @juliusvonkohout

**Pull request merged**